### ISSUE TITLE
SVGs load inconsistently in Docs mode for HTML stories when using file-loader

### ISSUE THREAD

**Issue by daneah** [storybookjs/storybook#9295]
**Describe the bug**
When authoring HTML stories and viewing them in Docs mode, when multiple stories contain an `<svg><use href="..."></svg>` only some of them (non-deterministically) render.

**To Reproduce**
1. Using `file-loader`, author a number of story previews in Docs mode that use `<svg>` with a `<use href="...">` reference
2. Run and open the Storybook
3. Navigate to Docs tab
4. Observe not all icons load

**Expected behavior**
Any correctly-specified SVGs render on the page

**Screenshots**
How the HTML stories render (sometimes more or different icons load):

<img width="250" alt="Screenshot 2020-01-01 21 48 56" src="https://user-images.githubusercontent.com/1808306/71649934-b6d61000-2ce0-11ea-83c5-d2339ea8a440.png">

How the React stories render, which is the expected behavior:

<img width="250" alt="Screenshot 2020-01-01 21 49 33" src="https://user-images.githubusercontent.com/1808306/71649930-afaf0200-2ce0-11ea-8cd8-3235f45ec4ec.png">

**Code snippets**
The HTML stories load the SVGs like so:

```jsx
const icons = require('@/assets/icons.svg');

<Preview>
  <Story name="Base alert">{`
    <svg role="img" aria-hidden="true">
      <use href="${icons}#status-base" />
    </svg>
  `</Story>
</Preview>

<Preview>
  <Story name="Info alert">{`
    <svg role="img" aria-hidden="true">
      <use href="${icons}#status-info" />
    </svg>
  `</Story>
</Preview>
```

**System:**
<details>
<summary>Environment info</summary>

```
  System:
    OS: macOS Mojave 10.14.5
    CPU: (12) x64 Intel(R) Core(TM) i7-8750H CPU @ 2.20GHz
  Binaries:
    Node: 12.4.0 - ~/.nvm/versions/node/v12.4.0/bin/node
    npm: 6.13.4 - ~/.nvm/versions/node/v12.4.0/bin/npm
  Browsers:
    Chrome: 79.0.3945.88
    Firefox: 71.0
    Safari: 12.1.1
  npmPackages:
    @storybook/addon-a11y: ^5.3.0-rc.6 => 5.3.0-rc.6
    @storybook/addon-docs: ^5.3.0-rc.6 => 5.3.0-rc.6
    @storybook/addon-knobs: ^5.3.0-rc.6 => 5.3.0-rc.6
    @storybook/addon-storysource: ^5.3.0-rc.6 => 5.3.0-rc.6
    @storybook/html: ^5.3.0-rc.6 => 5.3.0-rc.6
    @storybook/react: ^5.3.0-rc.6 => 5.3.0-rc.6
    @storybook/source-loader: ^5.3.0-rc.6 => 5.3.0-rc.6
    @storybook/vue: ^5.3.0-rc.6 => 5.3.0-rc.6
```
</details>

**Additional context**
We disable Storybook's default SVG loader and use `file-loader` instead. I might suspect this is related to the issue, except that the SVGs do in fact load, though sporadically. Does not exhibit issues in the React or Vue storybooks.

**Comment by shilman** [comment]
Sorry, missed this the first time around. I think the key difference might be that `@storybook/html` is rendering those stories in iframes, which can be problematic, whereas React is rendering them inline.

Vue also renders in iframes by default, but can be configured to run inline. But perhaps there are subtle rendering differences that cause it to be more reliable than `@storybook/html`.

There's also `@storybook/web-components` which can be configured to run inline and should support a superset of HTML. We could also probably use the same method to allow `@storybook/html` run inline. cc @daKmoR

**Comment by daneah** [comment]
Any thoughts on @shilman's commentary, @daKmoR?

**Comment by daneah** [comment]
Still an issue.

**Comment by daKmoR** [comment]
what is the output of that?

```
const icons = require('@/assets/icons.svg');

<Preview>
  <Story name="Base alert">{`
    <svg role="img" aria-hidden="true">
      <use href="${icons}#status-base" />
    </svg>
  `</Story>
</Preview>
```

where does it place the `defs`?
```
<use href="${icons}#status-base" />
```

I suspect that it only does a one time definition of those `defs` at the top and because in react it is on one page it is fine.

As soon as you introduce iframes (which get's used in html) it probably will not load as it can not access the `defs` in the main page...

so you probably would need to render the `defs` into every story...

PS: just a hunch 🙈

**Comment by daneah** [comment]
@daKmoR oooh that is an interesting thought, sounds plausible. Will look into that 😄

**Comment by daneah** [comment]
Sorry for the lack of follow-up here—we rapidly switched to web components only, so haven't really been digging into HTML Storybook at all, of late.

**Comment by vanessayuenn** [comment]
Hi! I am going through and cleaning up old issues in the repo. We just shipped a ton of improvements in Storybook 7.0, and I suspect this bug might be fixed already. Can you please try it out? If this issue persists with the latest version, please open a new bug report with a 7.x reproduction. Thank you so much! 🙏🏼
